### PR TITLE
third layout lookup

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
See #448 

**What did change**
Added a third layout lookup method 
+
When a sample cannot be found with esearch | efetch we change the sample name to its SRX and this seems to work for the edge-cases of @Rebecza 

**Checklist**
- [x] I made a PR to develop (not master)
- [x] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

**Comment by Maarten-vd-Sande** [comment]
@Rebecza can you test this?

Activate your seq2science environment and run
`pip install git+git://github.com/vanheeringen-lab/seq2science.git@hotfix-%23448`

Then check if sample GSM3832542 does not report any issues :)

**Comment by Maarten-vd-Sande** [comment]
sorry command doesnt work :(
`pip install "git+git://github.com/vanheeringen-lab/seq2science.git@378ff7e261fed4d8776eec9708b659f8b24eee95"`

**Comment by Maarten-vd-Sande** [comment]
Try again! 
`pip install "git+git://github.com/vanheeringen-lab/seq2science.git@739fe96db9aa147c4bdc574e50465c5d25482cf6`

**Comment by Rebecza** [comment]
No, still doesn't.

**Comment by Maarten-vd-Sande** [comment]
@siebrenf ready for review

File: seq2science/rules/configuration_generic.smk [hunk 1]
```diff
@@ -311,7 +337,8 @@ with FileLock(layout_cachefile_lock):
     config['layout'] = {**layout_cache,
                         **{k: v for k, v in config['layout'].items()},
                         **{k: v.get() for k, v in eutils_layout.items() if v.get() is not None},
-                        **{k: v.get() for k, v in trace_layout.items() if v.get() is not None}}
+                        **{k: v.get() for k, v in trace_layout1.items() if v.get() is not None},
+                        **{k: v.get() for k, v in trace_layout2.items() if v.get() is not None}}
```
Comment (siebrenf) [inline seq2science/rules/configuration_generic.smk:341]: trace_layout2 overwrites trace_layout1, which overwrites eutils_layout, which overwrites the layout_cache. Is that intended?

Comment (Maarten-vd-Sande) [reply seq2science/rules/configuration_generic.smk:341]: Yes that is intended (only overwriting when they are not None though). Trace layout has to overwrite eutils in case someone did an oopsie and wrote that their samples are single/paired end. Eutils is still necessary since it is the most robust method.
